pkg/build/build.go

```diff
@@ -317,6 +352,132 @@ func (ctx *Context) BuildWorkspace(workspaceDir string) error {
 	return nil
 }
 
+func copyFile(base, src, dest string, perm fs.FileMode) error {
+	basePath := filepath.Join(base, src)
+	destPath := filepath.Join(dest, src)
+	destDir := filepath.Dir(destPath)
+
+	inF, err := os.Open(basePath)
+	if err != nil {
+		return err
+	}
+	defer inF.Close()
+
+	if err := os.MkdirAll(destDir, 0o755); err != nil {
```

Comment:
Nit, for later:
If we're creating the directory, perhaps we should check access on the destination and also carry the same directory owner too?